[InstallDocs] Install markdowns added to 23.0-release (#6510)
aces/Loris: docs/wiki/00_SERVER_INSTALL_AND_CONFIGURATION/01_LORIS_Install/Ubuntu/README.md

`sudo apt install ...`.

* curl
* zip
* unzip
* php-json
* make
* software-properties-common
* php7.3-mysql
* php7.3-xml
* php7.3-mbstring
* php7.3-gd
* php7.3-zip
* php7.3-curl (for development instances only)
* libapache2-mod-php7.3

## Getting the source code

Visit the [releases](github.com/aces/loris/releases) page and download the zipped file of the latest LORIS
release. The below command will download the latest LORIS release and save 
it to a file named `loris-src.tar.gz`.

```bash
curl -s https://api.github.com/repos/aces/loris/releases/latest \
| grep 'tarball' \
| cut -d : -f2,3 \
| tr -d \" \
| tr -d ,\
| xargs -n 1 curl -s -o loris-src.tar.gz
```

When this is complete, expand the compressed file and move its contents to the
directory from which LORIS will be served. 

```bash
tar -zxf loris-src.tar.gz
```

This will create a folder called something like `aces-Loris-9e30cf0`. (The
part after `aces-Loris` is not important).

Rename this file to `loris`, e.g.

```bash
mv aces-Loris-9e30cf0 loris
```

Then move this folder to the web root (e.g. `/var/www/`) and go to this 
directory.

```bash
mv ./loris /var/www/; cd /var/www/loris

> 

Reviewer: ```suggestion
```
```
 line 80 needs 3 backticks to end formatting block

Reviewer 2: Editing the contents of these files will cause merge conflicts on master after the release.

Reviewer: yep https://github.com/aces/Loris/pull/6510#issuecomment-626045377

Reviewer: #6547

> ### If installing LORIS for production

Reviewer: Synthesize lines 81-85 so that the reader knows to run `make` on production and `make dev` on development

* software-properties-common
* php7.3-mysql
* php7.3-xml
* php7.3-mbstring
* php7.3-gd
* php7.3-zip
* php7.3-curl (for development instances only)
* libapache2-mod-php7.3

## Getting the source code

Visit the [releases](github.com/aces/loris/releases) page and download the zipped file of the latest LORIS
release. The below command will download the latest LORIS release and save 
it to a file named `loris-src.tar.gz`.

```bash
curl -s https://api.github.com/repos/aces/loris/releases/latest \
| grep 'tarball' \
| cut -d : -f2,3 \
| tr -d \" \
| tr -d ,\
| xargs -n 1 curl -s -o loris-src.tar.gz
```

When this is complete, expand the compressed file and move its contents to the
directory from which LORIS will be served. 

```bash
tar -zxf loris-src.tar.gz
```

This will create a folder called something like `aces-Loris-9e30cf0`. (The
part after `aces-Loris` is not important).

Rename this file to `loris`, e.g.

```bash
mv aces-Loris-9e30cf0 loris
```

Then move this folder to the web root (e.g. `/var/www/`) and go to this 
directory.

```bash
mv ./loris /var/www/; cd /var/www/loris
```

Once in the correct directory, run one of the following commands given your environment:

```bash

> make      // For production environments

Reviewer: ```suggestion
make     # For production environments
```
Looking at the new section following, and given this is bash -- let's use `#` instead of `//` for comments in this doc.

> make dev  // For development environments

Reviewer: ```suggestion
make dev  # For development environments
```

```

## Creating the lorisadmin user

> ```bash
> # Create lorisadmin user and group
> # Give lorisadmin `sudo` permission. This is required for the install process
> # in order to automatically generate Apache configuration files.
> # Sudo privileges can be revoked once the install is completed.
> sudo useradd -U -m -G sudo -s /bin/bash lorisadmin
> # Add apache to the lorisadmin group
> sudo usermod -a -G lorisadmin www-data
> # Set the password for the lorisadmin account
> sudo passwd lorisadmin
> sudo mkdir -m 755 -p /var/www/loris
> sudo chown lorisadmin.lorisadmin /var/www/loris
> su - lorisadmin
> ```

Reviewer: ```suggestion
Create lorisadmin user and group and give _lorisadmin_ `sudo` permission. This is required for several steps in the install process such as generating Apache configuration files. (Sudo privileges can be revoked once the install is completed.)
```bash
sudo useradd -U -m -G sudo -s /bin/bash lorisadmin
```
Next, add apache to the lorisadmin group
```bash
sudo usermod -a -G lorisadmin www-data
```
Then, set the password and login to the lorisadmin account
```bash
sudo passwd lorisadmin
su - lorisadmin
```
As a last step before running the install script --
Create the _loris root_ directory :
```bash
sudo mkdir -m 755 -p /var/www/loris
```
Note that if you do not perform the steps in the above order you may also have to run: 
```bash
sudo chown lorisadmin.lorisadmin /var/www/loris
```

## Running the install script

The next step in setting up LORIS is running the script `install.sh` in the 

> `tools/` directory. The script must be run from that directory.

Reviewer: ```suggestion
`tools/` directory. The script must be run from that directory, and must not be run using `sudo`.
```
note: this adds back the comment cut in john's changes.  Important to keep though because it still happens and will mess up your install.

This will begin an interactive setup process that will configure files and
permissions required to get LORIS up and running.

```bash

> cd tools/

Reviewer: ```suggestion
cd /var/www/tools/
```
because they haven't cd'd into it above.

* NodeJS 8.0 (or higher)
* composer

This guide does not cover installation of these requirements.

### Apt Packages
The following Ubuntu packages are required and should be installed using 
`sudo apt install ...`.

* curl
* zip
* unzip
* php-json
* make
* software-properties-common
* php7.3-mysql
* php7.3-xml
* php7.3-mbstring
* php7.3-gd
* php7.3-zip
* php7.3-curl (for development instances only)
* libapache2-mod-php7.3

## Getting the source code

Visit the [releases](github.com/aces/loris/releases) page and download the zipped file of the latest LORIS
release. The below command will download the latest LORIS release and save 
it to a file named `loris-src.tar.gz`.

```bash
curl -s https://api.github.com/repos/aces/loris/releases/latest \
| grep 'tarball' \
| cut -d : -f2,3 \
| tr -d \" \
| tr -d ,\
| xargs -n 1 curl -s -o loris-src.tar.gz
```

When this is complete, expand the compressed file and move its contents to the
directory from which LORIS will be served. 

```bash
tar -zxf loris-src.tar.gz
```

This will create a folder called something like `aces-Loris-9e30cf0`. (The
part after `aces-Loris` is not important).

> Rename this file to `loris`, e.g.

Reviewer: ```suggestion
Rename this to `loris`, e.g.
```

```bash
mv aces-Loris-9e30cf0 loris
```

Then move this folder to the web root (e.g. `/var/www/`) and go to this 
directory.

```bash
mv ./loris /var/www/; cd /var/www/loris
```

Once in the correct directory, run one of the following commands given your environment:

```bash
make      # For production environments
make dev  # For development environments
```

## Creating the lorisadmin user
Create the _lorisadmin_ user and group and give _lorisadmin_ `sudo` permission. 
This is required for the install process in order to automatically generate
Apache configuration files. Sudo priviledges can be revoked once the install
is completed. 

```bash
sudo useradd -U -m -G sudo -s /bin/bash lorisadmin
```

Next, add Apache to the _lorisadmin_ group

```bash
sudo usermod -a -G lorisadmin www-data
```
Then, set the password for the _lorisadmin_ account

```bash
sudo passwd lorisadmin
su - lorisadmin
```

Finally, create the _loris root_ directory

```bash
sudo mkdir -m 755 -p /var/www/loris
sudo chown lorisadmin.lorisadmin /var/www/loris
```

## Running the install script

The next step in setting up LORIS is running the script `install.sh` in the 

> `tools/` directory. The script must be run from that directory, and must not be

Reviewer: ```suggestion
`tools/` directory. The script must be run from that directory, and must _not_ be
```

> run using `sudo`.

Reviewer: ```suggestion
run using sudo.
```

# Installing LORIS In-Depth

This guide assumes an installation on an Ubuntu server. 

If you are using CentOS, please visit the CentOS installation guide for 
instructions on installation.

When you've completed this guide, you should be able to load and log into LORIS in your browser. Further setup to customize LORIS for your project will be required after that. Please visit
[the Setup page](https://github.com/aces/Loris/wiki/Setup) in order to complete the setup for LORIS.

## Prequisities

### Server Stack

LORIS requires a LAMP stack in order to run, specifically:

> * Apache 2.4

Reviewer 2: Since we're modifying these files a bit, can you space out these lists so that they render properly? The CentOS one has the same problem. See #6623

* Apache 2.4  

* MySQL 5.7 (or MariaDB 10.3) (or higher)  

* PHP 7.3 (or higher)  

Additionally, the following package manager are required to build LORIS:  

* NodeJS 8.0 (or higher)  

* composer

This guide does not cover installation of these requirements.

### Apt Packages
The following Ubuntu packages are required and should be installed using 
`sudo apt install ...`.

* curl  

* zip  

* unzip  

* php-json  

* make  

* software-properties-common  

* php7.3-mysql  

* php7.3-xml  

* php7.3-mbstring  

* php7.3-gd  

* php7.3-zip  

* php7.3-curl (for development instances only)  

* libapache2-mod-php7.3  

## Getting the source code

Visit the [releases](https://github.com/aces/loris/releases) page and download the zipped file of the latest LORIS
release. The below command will download the latest LORIS release and save 
it to a file named `loris-src.tar.gz`.

```bash
curl -s https://api.github.com/repos/aces/loris/releases/latest \
| grep 'tarball' \
| cut -d : -f2,3 \
| tr -d \" \
| tr -d ,\
| xargs -n 1 curl -L -s -o loris-src.tar.gz
```

When this is complete, expand the compressed file and move its contents to the

> directory from which LORIS will be served. 

Reviewer: ```suggestion
_loris root directory_ from which LORIS will be served: 
```

```bash
tar -zxf loris-src.tar.gz
```

This will create a folder called something like `aces-Loris-9e30cf0`. (The
part after `aces-Loris` is not important).

Rename this folder to `loris`, e.g.

```bash
mv aces-Loris-9e30cf0 loris
```

Then move this folder to the web root (e.g. `/var/www/`) and go to this 
directory.

```bash

> mv ./loris /var/www/; cd /var/www/loris

Reviewer: ```suggestion
mv ./loris /var/www/
cd /var/www/loris
```
because there's no need to put them on line and the `./loris` part is key to observe

```bash
mv ./loris /var/www/; cd /var/www/loris
```

### If installing LORIS for production

```bash
cd `/var/www/loris/`; make
```

### If installing LORIS for development

```bash

> cd `/var/www/loris/`; make

Reviewer: ```suggestion
cd `/var/www/loris/`; make dev
```

mv ./loris /var/www/; cd /var/www/loris
```

> Once in the correct directory, run the following commands:

Reviewer: ```suggestion
Once in the correct directory, run one of the following commands given your environment:
```

make dev  // For development environments
```

## Creating the lorisadmin user
```bash
# Create lorisadmin user and group
# Give lorisadmin `sudo` permission. This is required for the install process
# in order to automatically generate Apache configuration files.
# Sudo privileges can be revoked once the install is completed.
sudo useradd -U -m -G sudo -s /bin/bash lorisadmin
# Add apache to the lorisadmin group
sudo usermod -a -G lorisadmin www-data
# Set the password for the lorisadmin account
sudo passwd lorisadmin

> sudo mkdir -m 755 -p /var/www/$projectname

Reviewer: let's make the root directory consistent across this readme. i.e. `/var/www/loris`

This will begin an interactive setup process that will configure files and
permissions required to get LORIS up and running.

> The install script will prompt for the user password in order to use `sudo` permission for configuring

Reviewer: Why remove this paragraph?   Definitely keep it, is what I'd suggest

Author: This was removed in the PR #6181 by John. I believe it was removed because the above addition grants sudo privileges, so these instructions are not necessary. Let me know what you think.

Your LORIS instance should now be accessible by pointing your browser URL to `http://%IPADDRESS%`.

> If you use MySQL 8 and PHP 7.3 (instead of PHP 7.4), please read [this link](https://www.php.net/manual/en/mysqli.requirements.php) and also [this](https://dev.mysql.com/doc/refman/8.0/en/upgrading-from-previous-series.html#upgrade-caching-sha2-password-compatible-connectors)

Reviewer 2: I think this should go into #6511 instead of this PR.
